### ISSUE TITLE
Typedefs for arrays are always resolved

### ISSUE THREAD

**Issue by k0ekk0ek** [eclipse-cyclonedds/cyclonedds-cxx#114]
Array specifiers are always resolved because `get_cpp11_type` is used in the generator, even if it's a `typedef`. e.g.

```
typedef boolean barr[2];

struct foo {
  barr bar;
};
```

In this case the `struct` member uses `std::array<boolean, 2>`, while `barr` is expected. I've looked into it a bit, and while trivial, it does require quite a bit of code to be modified. I propose we fix it after the 0.8.0 release.

**Comment by Bit-Seq** [comment]
Hello k0ekk0ek,

Has there been any change with this issue? I have been trying to understand this issue and after reviewing it I have reproduced what is described on it.

Following your example (`HelloWorldData.idl`):

```
  typedef boolean barr[2];

  struct Msg {
    barr bar;
  };
```

Is converted to (`HelloWorldData.hpp`):

```
private:
 std::array<bool, 2> bar_ = { };

public:
  Msg() = default;

  explicit Msg(
    const std::array<bool, 2>& bar) :
    bar_(bar) { }
```

Isn't that correct? The conversion from idlc to C++ seems to be correct.

I have also realised that the description says it was going to be fixed after `cyclonedds/cyclonedds-cxx` 0.8.0 version. If I am not in a mistake the current version is the 0.10.0. So, that is the reason of I have the doubt if perhaps that was resolved, or I am not understanding the problem.

Thanks and regards.

**Comment by eboasson** [comment]
Hi @Bit-Seq, this indeed seems to be in accordance with the IDL-to-C++11 mapping. I have also checked that it has the same effect as:
```C++
private:
 barr bar_ = { };

public:
  Msg() = default;

  explicit Msg(
    const barr& bar) :
    bar_(bar) { }
```
which is what I had (naively) expected to be generated.

In short, it looks fixed to me. Thanks for asking!
